### ISSUE TITLE
[Bug] LWC 3.8 : Children attribute table not displayed 

### ISSUE THREAD

**Issue by Alain-LG** [3liz/lizmap-web-client#4935]
### What is the bug? (in English)

Since I've been using LWC version 3.8, I've lost the child attribute table.
I can see the parent's attribute table and the button but when I choose a parent the list of children is not displayed.

It was OK with LWC versions 3.7 and 3.6

### Steps to reproduce the issue

1. I have declared a relation between my table "Suivi marché" and "tronçons"
2. I have plublished them by WFS
3. I have added them in the attribute table tab
4. When I show the parents'attribute table, Its OK.
5. When I click on a parent, the list of children is not diplayed.

![image](https://github.com/user-attachments/assets/642acb96-b4d2-4b26-aa39-934effc70e72)

### Versions, safeguards, check summary etc

**Versions :**

* Lizmap Web Client : 3.8.2
* Lizmap plugin : 4.4.4
* QGIS Desktop : 3.34.11
* Lizmap.com : Yes
* QGIS Server : 3.34.11
* Py-QGIS-Server : 1.9.1
* QGIS Server plugin DataPlotly : Not installed
* QGIS Server plugin atlasprint : 3.4.1
* QGIS Server plugin cadastre : 1.20.0
* QGIS Server plugin lizmap_server : 2.11.2
* QGIS Server plugin wfsOutputExtension : 1.8.2

<details>
<summary>List of Lizmap Web Client modules :</summary>
<br/>
* saas : 1.7.4
* webdav : 1.1.6
</details>

<details>
<summary>List of safeguards :</summary>
<br/>
* Mode : normal<br/>
* Allow parent folder : yes<br/>
* Number of parent : 2 folder(s)<br/>
* Prevent other drive : yes<br/>
* Prevent PG service : yes<br/>
* Prevent PG Auth DB : yes<br/>
* Force PG user&pass : yes<br/>
* Prevent ECW : yes<br/>
</details>

### Check Lizmap plugin

- [X] I have done the step just before in the Lizmap QGIS desktop plugin before opening this ticket. Otherwise, my ticket is not considered valid and might get closed.

### Operating system

Hébergement lizmap

### Browsers

Firefox

### Browsers version

128.0

### Relevant log output

```shell
No log output to share.
```

**Comment by Alain-LG** [comment]
I agree with you, It's OK with a copy of your "cats demo" project on my own instance hosted on your servers. I tried this to know if it was an installation problem or if it comes from my project...

Here is my URL : [https://agglofecampcauxlittoral.lizmap.com/test/index.php/view/map?repository=entretienev&project=Marche_entretien]
You can use this login/pwd to connect : marche_entretien/marche_entretien

I don't understad why my childrens attribute table is not displayed whereas I think I have done the same configuration.

I'm currently looking for a solution by changing my foreign key name to lowercase. As a result, I have other small problems due to this field name change.
This foreign key is declared as a relationship reference in the QGIS desktop, could it be the cause of this trouble ?

Thanks a lot for your help.

**Comment by Alain-LG** [comment]
I found the reason why it wasn't working: my layer names contain spaces.

The solution is therefore to rename the layer in the layer tree and modify the title of these layers in the metadata.

**Comment by Gustry** [comment]
> The solution is therefore to rename the layer in the layer tree and modify the title of these layers in the metadata.

So there is an issue.
But there are a few corner cases, like if your layer has a shortname etc.

Can you replicate the issue on a project ?

**Comment by Gustry** [comment]
I'm not sure if the space is the real problem. I just added some test data in #4942 , I couldn't replicate with only the space difference. I haven't debug more for now.

**Comment by vlebert** [comment]
Hi @Gustry 
I encountered the same issue. Deleting spaces in child layer did solve the issue un LWC.
What is strange is that when opening Dev Tool, when clicking on a parent feature, the child feature were received by the browser, but not displayed in the child table element

**Comment by vlebert** [comment]
It is an issue though because it was previously working with space, so some published project were broken by some update
